js/private/node-patches/src/fs.ts

```diff
@@ -621,19 +642,16 @@ export const patcher = (fs: any = _fs, roots: string[]) => {
         function _tryParent(): boolean {
             nested.push(path.basename(maybe))
             const dirname = path.dirname(maybe)
-            if (
-                !dirname ||
-                dirname == maybe ||
-                dirname == '.' ||
-                dirname == '/'
-            ) {
+            // TODO: also look for paths that are not within any 'roots'
```

Comment:
Both of these TODOs were causing test failures that I don't understand atm so I'm just going to do what I have below for now.